Topics/Tech_Stacks/Tanstack_Table.md

```diff
@@ -0,0 +1,141 @@
+| Poodle          | Charlie      | 1.6         |
+| Bulldog         | Daisy        | 1.0         |
+| Chihuahua       | Rocky        | 0.8         |
+| Siberian Husky  | Coco         | 1.4         |
+
+Luckily, this is easy to do with Typescript. 
+
+```typescript
+// Define the shape of your columns
+type Dog = {
+    dogBreed: string
+    dogName: string
+    dogSize: number
+}
+
+// Converts your typescript type into an object that the library can use effectively
+const columnHelper = createColumnHelper<Dog>()
+
+// Tells the library how you should render the columns
+// If you want columns to be grouped together you can define it here as well. 
+// See: https://tanstack.com/table/v8/docs/guide/grouping
+const defaultColumns = [
+  // Display Column, the accessor() function takes the name of the attribute you want to display.
+  columnHelper.accessor("dogBreed", {
+    header: "Dog Breed",
+    cell: (info) => info.getValue(),
+  }),
+  columnHelper.accessor("dogName", {
+    header: "Dog Name",
+    cell: (info) => info.getValue(),
+  }),
+  columnHelper.accessor("dogSize", {
+    header: "Dog Size",
+    cell: (info) => info.getValue(),
+  })
+]
+
+// Follows the same format as Dog[]
+const exampleData = [
+    {
+        "dogBreed": "Labrador",
+        "dogName": "Bella",
+        "dogSize": 1.0
+    },
+    {
+        "dogBreed": "Beagle",
+        "dogName": "Max",
+        "dogSize": 1.2
+    },
+    {
+        "dogBreed": "German Shepherd",
+        "dogName": "Luna",
+        "dogSize": 2.0
+    },
+]
+
+```
+
+Next, we can begin writing our component.
+
+```tsx
+import { useReactTable } from '@tanstack/react-table'
+
+/* Insert code from above */
+
+function DogTable() {
+  const table = useReactTable({
+    columns,
+    data
+    // You can insert other options you want to add to your table, such as being able to resize, the columns, sorting methods, etc.
+  })
+  
+  return (
+    <table>
+        <thead>
+            {table.getHeaderGroups().map(headerGroup => (
+                <tr key={headerGroup.id}>
+                    {headerGroup.headers.map(header => (
+                    <th key={header.id}>
+                        {flexRender(
+                            header.column.columnDef.header,
+                            header.getContext()
+                        )}
+                    </th>
+                    ))}
+                </tr>
+            ))}
+        </thead>
+        <tbody>
+            {table.getRowModel().rows.map(row => (
+                <tr key={row.id}>
+                    {row.getVisibleCells().map(cell => (
+                    <td key={cell.id}>
+                        {flexRender(
+                            cell.column.columnDef.cell,
+                            cell.getContext()
+                        )}
+                    </td>
+                    ))}
+                </tr>
+            ))}
+        </tbody>
+    </table>
+  )
+}
+```
```

Comment:
you should add a little bit of a conclusion to wrap up on what you did here, maybe some explanations on something you're using implicitly, like context.

Reply:
Thank you, that's an excellent idea. I've done so in the next draft